### PR TITLE
refactor CANParser to improve performance

### PR DESCRIPTION
**Pull request by deanlee**
1. pass by reference and use vector::emplace_back to reduce memory copies
2. Added new function `CANParser::update_strings` to parse all can events in the strings array, and only call `query_latest` once to update the latest data.  the code in cython also only needs to parse the latest message once. this should greatly improve performance when dealing with multiple can messages.
3. Only expose function `CANParser::update_strings` to cython.

### CODE REVIEW

**Comment by deanlee** [comment]
looks like the following code is the performance killer.
```
for v in self.vl_all.values():
      v.clear()

```

**Comment by deanlee** [comment]
It looks like half of the time is spent executing the following statement:

> 
>     for v in self.vl_all.values():
>       v.clear()

**Comment by deanlee** [comment]
the time used for parse all signals in message:

|   | before | after |
| ------------- | ------------- | ------------- |
| update_string  | 687.6ms to parse 50000, avg: 13751ns  | 543.6ms to parse 50000, avg: 10871ns |
| update_strings  | 409.5ms to parse 50000, avg: 8190ns  | 85.0ms to parse 50000, avg: 1699ns |

update_string doesn't improve much because we calls update_strings([m]) in loop to  implement the testing for update_string. this will create a new list every time.

**Comment by deanlee** [comment]
@sshane I have not changed the parameters for `CANParser::CANParser`. you can change it to what you want.

**Comment by sshane** [comment]
We can do that in another PR if this one is good to go standalone!

File: can/parser_pyx.pyx [hunk 1]
```diff
@@ -101,47 +102,34 @@ cdef class CANParser:
       message_options_v.push_back(mpo)
 
     self.can = new cpp_CANParser(bus, dbc_name, message_options_v, signal_options_v)
-    self.update_vl()
+    self.update_strings([])
 
-  cdef unordered_set[uint32_t] update_vl(self):
+  def update_string(self, dat, sendcan=False):
```
Comment (adeebshihadeh) [inline can/parser_pyx.pyx:107]: let's get rid of the cython update_string. it's not used anywhere

**Review by adeebshihadeh — COMMENTED** [review]
Nice speedup and cleanup!

File: can/parser_pyx.pyx (hunk 1 above)
Comment (deanlee) [reply can/parser_pyx.pyx:107]: it's still used in can/tests/test_packer_parser.py and can/tests/test_parser.py.

should we replace it with `update_strings` in these two files? or keep update_string as it is?

Comment (adeebshihadeh) [reply can/parser_pyx.pyx:107]: let's replace it

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -14,6 +14,10 @@ jobs:
   unit-tests:
     name: unit tests
     runs-on: ubuntu-20.04
+    #strategy:
```
Comment (adeebshihadeh) [inline .github/workflows/tests.yml:17]: @deanlee in the future, you can uncomment this to run it several times

Comment (deanlee) [reply .github/workflows/tests.yml:17]: ok!

**Pull request merged**